docs/source/udf.rst

```diff
@@ -7,6 +7,42 @@ of LiberTEM allows users to run their own reduction functions easily, without ha
 to worry about parallelizing, I/O, the details of buffer management and so on. 
 
 
+How UDF works in layman's terms
+-----------------------------------
+
+.. image:: ./images/diagram.svg
+
+The UDF interface of LiberTEM is heavily utilizing the existing LiberTEM architecture. First,
+data is partitioned into several `"partitions"` and distributed across workers. Then, each partition,
+which can be viewed as a collection of frames, are processed by the user-defined `process_frame` function.
+Here, frames are fetched in an iterative manner, and one could view `process_frame` as `merging` the previous
+computations (i.e., the result computed on previously considered set of frames) and the newly added frame.
```

Comment:
Hmm, it kind of depends on the operation. If you only have `kind="nav"` buffers, you can view `process_frame` as working independently on different frames. For `kind="sig"`, this merge-inside-`process_frame` description is spot on, though!

Also, this "merging" inside of `process_frame` needs to be differentiated from the `merge` method described below: `process_frame` only has access to the results from inside the current partition, whereas `merge` always merges the results from a single partition into the full result. (maybe  "previously considered set of frames of a partition" or something like this?)